### ISSUE TITLE
Cannot use default value with class alias

### ISSUE THREAD

**Issue by sgalles** [ceylon/ceylon-compiler#2167]
This code
```ceylon
class Answer(Integer i = 42) => Integer(i);

shared void run() { 
    print(Answer());
}
```
compiled
```
source\org\boo\run.ceylon:8: error: constructor Integer in class Integer cannot be applied to given types;
    print(Answer());
                ^
  required: long
  found: no arguments
  reason: actual and formal argument lists differ in length
```

**Comment by FroMage** [comment]
I seem to recall deciding that aliases could not define default values in the past, no? @tombentley @gavinking ?

**Comment by tombentley** [comment]
Well my only recollection is, basically, that we have a big TODO for class aliases (https://github.com/ceylon/ceylon-compiler/issues/1295) which has never been high enough priority to actually get implemented. So I think you're right, and we kinda decided it wasn't allowed until we fix that and then it should be allowed.

**Comment by gavinking** [comment]
Well there's also ceylon/ceylon-spec/issues/736.

**Comment by FroMage** [comment]
Indeed. So we should just disallow this for now in the typechecker, it's totally not trivial and does not seem very useful when the same can be achieved with a function rather than an alias. @gavinking ?

**Comment by FroMage** [comment]
Added ceylon/ceylon-spec#1333 for the typechecker and moving to 1.3
